File: ABC/001-050/027/C/main.py
n = int(input())
# x -> 2x or 2x + 1
# if x > n -> break
# 最後に取った人が負け
# 先攻Takahashi , Aoki

x = 1
turn = False
while True:
    saizen = 0
    one = 2 * x
    two = 2 * x + 1
    print(one, two, n)
    # 殺す選択を取る
    if (2 * (two)) > n:
        saizen = two
    # elif (2 * (two) + 1) > n:
    #     if 2 * (2 * (one) + 1) <= n:
    #         saizen = one
    # print(saizen)

    # 死なない選択をとる
    elif 2 * (2 * (two) + 1) <= n:
        saizen = two
    elif 2 * (2 * (two) + 1) > n:
        if 2 * (2 * (one) + 1) <= n:
            saizen = one
    x = saizen

    if x > n or saizen == 0:
        if turn == False:
            print('Aoki')
            break
        else:
            print('Takahashi')
            break
    turn = not turn

File: ABC/001-050/027/C/main.c
#include<stdio.h>

// 3 -> 右，左
// 4 -> 左，右

int main(void){
  long long num;
  int depth = 0;
  long long x = 1;
  scanf("%lld", &num);
  for(long long n = num; n > 0; n /= 2)
    depth++;
  // printf("%d\n", depth);
  if(depth % 2 == 0){
    int turn = 0;
    while(1){
      if(turn == 0){
        x = x * 2;
        if(x > num){
          printf("Aoki\n");
          break;
        }
        turn++;
      } else {
        x = x * 2 + 1;
        if(x > num){
          printf("Takahashi\n");
          break;
        }
        turn--;
      }
    }
  } else {
    int turn = 0;
    while(1){
      if(turn == 0){
        x = x * 2 + 1;
        if(x > num){
          printf("Aoki\n");
          break;
        }
        turn++;
      } else {
        x = x * 2;
        if(x > num){
          printf("Takahashi\n");
          break;
        }
        turn--;
      }
    }
  }
}
